src/noit_metric_tag_search.c

```diff
@@ -1373,8 +1354,27 @@ noit_metric_tag_search_evaluate_against_metric_id(const noit_metric_tag_search_a
     return mtev_false;
   }
 
-  const noit_metric_tagset_t *tagsets[3] = { &tagset_check, &tagset_stream, &tagset_measurement };
-  mtev_boolean ok = noit_metric_tag_search_evaluate_against_tags_multi(search, tagsets, 3);
+  // setup implicit tags
+  noit_metric_tagset_t tagset_implicit;
+  MKTAGSETCOPY(tagset_implicit);
+  char implicit_str[NOIT_IMPLICIT_TAG_MAX_PAIR_LEN + 1 +
+                    sizeof("__check_uuid:") + UUID_PRINTABLE_STRING_LENGTH];
+  size_t uuid_index =
+      snprintf(implicit_str, sizeof(implicit_str),
+               "__name:%.*s,__check_uuid:", id->name_len, id->name);
+  mtev_uuid_unparse_lower(id->id, implicit_str + uuid_index);
```

Comment:
``mtev_uuid_unparse_lower`` does not null terminate the output, is that ok here?

Reply:
yes. `noit_metric_add_implicit_tags_to_tagset` uses a `size_t` to determine the end of the tag string. In this case, the size is the return of `snprintf` (the size of the formatted string) plus `UUID_PRINTABLE_STRING_LENGTH` (the size of the uuid appended by `mtev_uuid_unparse_lower`.